### PR TITLE
chore(node): improve node/binding structure

### LINKED ISSUE #432: `yarn build --no-wasm` could not detect changes of `rolldown_binding`'s dependecies
To make sure the [`just test-node`](https://github.com/rolldown-rs/rolldown/pull/431/files#diff-deb9bb56fb122db0b605aa5b63f95a4665c905b18dd670e1fa6c877576a94ff1R49) always running node tests with latest code, `yarn build --no-wasm` need to build `rolldown_node_binding` freshly when there are changes of `rolldown_binding`'s dependecies.

@milesj I'm not sure how to fix this behavior, maybe watch all files of `crates/**/*.rs`?

### LINKED ISSUE #545: Improve Node package / binding structure
Currently `@rolldown/node` and `@rolldown/node-binding` are two separate packages with src in two different locations (`packges/node` and `crates/rolldown_binding`).

Now that we have acquired the `rolldown` npm package, the plan is to merge both into a single npm package.

- This package will contain both the CLI and directly depends on platform-specific bindings as optional dependencies (including the wasm binding)
- src for this package will be in `packages/rolldown`
- `crates/rolldown_binding` needs to adjust napi-rs build to target `packages/rolldown`
- `@rolldown/node` and `@rolldown/node-binding` packages will be deprecated as they are no longer needed

### PR DESCRIPTION
**Pull request by hyf0**
### Description

Closes #545.

> I see `"@napi-rs/cli": "^3.0.0-alpha.43"` is smart enough to determine whether to rebuild or not if changes don't contain rust files.

Closes #432. 

### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
* **#550** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/550?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by hyf0** [comment]
Since this is a big folder structure change and the ci run successfully, let's merge it first and fix possible issue on top of it.

File: packages/rolldown/package.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "aarch64-apple-darwin",
+      "aarch64-linux-android",
+      "x86_64-unknown-freebsd",
+      "aarch64-unknown-linux-musl",
+      "aarch64-pc-windows-msvc",
+      "armv7-linux-androideabi"
+    ]
+  },
+  "dependencies": {
+    "prettier": "^3.2.5"
```
Comment (yyx990803) [inline packages/rolldown/package.json:65]: This is by mistake right?

File: packages/rolldown/package.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "rolldown",
+  "version": "0.0.1",
```
Comment (yyx990803) [inline packages/rolldown/package.json:3]: The package previously has versions published up to 0.3.0 and 0.4.x-alpha, we should start at 0.5.0 just to be safe.

File: packages/rolldown/package.json (hunk 1 above)
Comment (hyf0) [reply packages/rolldown/package.json:65]: Yeah.

**Comment by yyx990803** [comment]
@hyf0 found some small issues. Pinging in case the inline comments get lost.

**Pull request merged**